ts_scripts/install_dependencies.py

```diff
@@ -190,55 +241,46 @@ def __init__(self):
         self.sudo_cmd = "" if os.geteuid() == 0 else self.sudo_cmd
 
         if args.force:
-            os.system(f"{self.sudo_cmd}apt-get update")
+            subprocess.run([self.sudo_cmd, "apt-get", "update"], check=True)
 
     def install_java(self):
-        if os.system("javac --version") != 0 or args.force:
-            os.system(f"{self.sudo_cmd}apt-get install -y openjdk-17-jdk")
+        if self.check_command("javac --version") or args.force:
+            subprocess.run([self.sudo_cmd, "apt-get", "install", "-y", "openjdk-17-jdk"], check=True)
 
     def install_nodejs(self):
-        if os.system("node -v") != 0 or args.force:
-            os.system(
-                f"{self.sudo_cmd}curl -sL https://deb.nodesource.com/setup_18.x | {self.sudo_cmd}bash -"
-            )
-            os.system(f"{self.sudo_cmd}apt-get install -y nodejs")
+        if self.check_command("node -v") or args.force:
+            subprocess.run([self.sudo_cmd, "curl", "-sL", "https://deb.nodesource.com/setup_18.x", "|", "bash", "-"], check=True)
+            subprocess.run([self.sudo_cmd, "apt-get", "install", "-y", "nodejs"], check=True)
 
     def install_wget(self):
-        if os.system("wget --version") != 0 or args.force:
-            os.system(f"{self.sudo_cmd}apt-get install -y wget")
+        if self.check_command("wget --version") or args.force:
+            subprocess.run([self.sudo_cmd, "apt-get", "install", "-y", "wget"], check=True)
 
     def install_numactl(self):
-        if os.system("numactl --show") != 0 or args.force:
-            os.system(f"{self.sudo_cmd}apt-get install -y numactl")
+        if self.check_command("numactl --show") or args.force:
+            subprocess.run([self.sudo_cmd, "apt-get", "install", "-y", "numactl"], check=True)
 
     def install_cpp_dependencies(self):
-        os.system(
-            f"{self.sudo_cmd}apt-get install -y {' '.join(CPP_LINUX_DEPENDENCIES)}"
-        )
+        subprocess.run([self.sudo_cmd, ["apt-get", "install", "-y"] + list(CPP_LINUX_DEPENDENCIES)], check=True)
 
     def install_neuronx_driver(self):
         # Configure Linux for Neuron repository updates
-        os.system(
-            ". /etc/os-release\n"
-            + f"{self.sudo_cmd}tee /etc/apt/sources.list.d/neuron.list > /dev/null <<EOF\n"
-            + "deb https://apt.repos.neuron.amazonaws.com ${VERSION_CODENAME} main\n"
-            + "EOF\n"
-            + "wget -qO - https://apt.repos.neuron.amazonaws.com/GPG-PUB-KEY-AMAZON-AWS-NEURON.PUB | sudo apt-key add -"
-        )
+        subprocess.run([self.sudo_cmd, "tee", "/etc/apt/sources.list.d/neuron.list"], input=f"deb https://apt.repos.neuron.amazonaws.com {os.environ['VERSION_CODENAME']} main\n", text=True, check=True)
+        subprocess.run(["wget", "-qO", "-", "https://apt.repos.neuron.amazonaws.com/GPG-PUB-KEY-AMAZON-AWS-NEURON.PUB"], check=True)
```

Comment:
The piping of the output from wget to apt-get will be missing

Reply:
Fixed.